client/src/api/v1/index.ts

```diff
@@ -7,14 +11,21 @@ export default class ServerApi {
     this.api = axios.create({ baseURL: `api/v1` });
   }
 
-  protected async post<Type>(path: string, form: Type) {
+  protected async post<T, R>(
+    path: string,
+    form: T
+  ): Promise<{ status: number; data?: R }> {
     try {
       const response = await this.api.post(path, form);
-      return { response: response };
+      return { status: response.status, data: response.data };
     } catch (error) {
       if (axios.isAxiosError(error)) {
-        return { error: error };
+        return {
+          status: error.response?.status as number,
+          data: error.response?.data,
```

Comment:
In the instance of a failure, there wouldn't be any data which conforms to the type of the return.

Reply:
right, ill remove the optional indicator from the return type since we actually want it to always return data, even in error (the data contains the error message)